src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.cs

```diff
@@ -93,8 +93,21 @@ protected override void FreeNotFirstVisitor(AbstractSymbolDisplayVisitor visitor
 
         internal SymbolDisplayPart CreatePart(SymbolDisplayPartKind kind, ISymbol? symbol, string text)
         {
-            text = (text == null) ? "?" :
-                   (_escapeKeywordIdentifiers && IsEscapable(kind)) ? EscapeIdentifier(text) : text;
+            if (text == null)
+            {
+
+                return new SymbolDisplayPart(kind, symbol, "?");
+            }
+
+            if (!_escapeKeywordIdentifiers)
+            {
+                return new SymbolDisplayPart(kind, symbol, text);
+            }
+
+            bool escapeRecord = StringComparer.Ordinal.Equals(text, "record") && symbol?.Kind is SymbolKind.NamedType or SymbolKind.Alias;
```

Comment:
It feels like the actual text comparison is better in `EscapeIdentifier`. I would simply pass along a flag covering whether the text is a namedtype or alias name, and let `EscapeIdentifier` handle actually looking at the text.